cypress/tests/ui/menu/menu.cy.js

```diff
@@ -1 +1,172 @@
+        expect(queryParamValue.toLowerCase()).to.be.eq(
+          menuOptionValue.toLowerCase()
+        );
+      });
+
+      // Pipeline Label in the Menu
+      cy.get('.pipeline-nodelist__row__label')
+        .first()
+        .invoke('text')
+        .then((pipelineLabel) => {
+          expect(menuOptionValue.toLowerCase()).to.include(
+            pipelineLabel.toLowerCase()
+          );
+        });
+    });
+  });
+
+  it('verifies that users can search/filter for a flowchart component using the search box. #TC-17', function () {
+    const searchInput = 'Ingestion';
+    cy.get('.search-input__field').type(searchInput);
+
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(200);
+
+    // Pipeline Label in the Menu
+    cy.get('.pipeline-nodelist__row__label')
+      .first()
+      .invoke('text')
+      .then((pipelineLabel) => {
+        expect(searchInput.toLowerCase()).to.include(
+          pipelineLabel.toLowerCase()
+        );
+      });
+  });
+
+  it('verifies that users can select a node/dataset/parameters from the menu. #TC-18', function () {
+    const nodeToClickText = 'Companies';
+
+    // Action
+    cy.get(`[data-test=node-${nodeToClickText}]`)
+      .should('exist')
+      .as('nodeToClick');
+    cy.get('@nodeToClick').click();
+
+    // Assert after action
+    cy.__checkForText__(
+      '.pipeline-node--selected > .pipeline-node__text',
+      nodeToClickText
+    );
+    cy.__checkForText__('.pipeline-metadata__title', nodeToClickText);
+  });
+
+  it('verifies that users can highlight a node/dataset/parameters from the menu by hovering on their name. #TC-19', function () {
+    const nodeToHighlightText = 'Companies';
+
+    // Action
+    cy.get(`[data-test=node-${nodeToHighlightText}]`)
+      .should('exist')
+      .as('nodeToHighlight');
+    cy.__hover__('@nodeToHighlight');
+
+    // Assert after action
+    cy.__checkForText__(
+      '.pipeline-node--active > .pipeline-node__text',
+      nodeToHighlightText
+    );
+  });
+
+  it('verifies that users can hide/show a node/dataset/parameters in the flowchart, by clicking on the eye icon. #TC-20', function () {
+    const nodeToToggleText = 'Companies';
+
+    // Alias
+    cy.get(`.pipeline-nodelist__row__checkbox[name=${nodeToToggleText}]`).as(
+      'nodeToToggle'
+    );
+
+    // Assert before action
+    cy.get('@nodeToToggle').should('be.checked');
+    cy.get('.pipeline-node__text').contains(nodeToToggleText);
+
+    // Action
+    cy.get('@nodeToToggle').uncheck({ force: true });
+
+    // Assert after action
+    cy.__checkForText__(
+      `[data-test=node-${nodeToToggleText}] > .pipeline-nodelist__row__label--faded`,
+      nodeToToggleText
+    );
+    cy.get('.pipeline-node__text').should('not.contain', nodeToToggleText);
+  });
+
+  it('verifies that users can select and only show a node/dataset/parameters in the flowchart, by clicking on the box icon. #TC-21', function () {
```

Comment:
i know the user flow have it like that. but i think it should 
```suggestion
  it('verifies that users can select and only show a node/dataset/parameters in the flowchart, by clicking on the focus mode. #TC-21', function () {
```

Reply:
also i was thinking the most obvious visible change when you click on focus mode is that only that pipeline shows maybe for assertion we can assert the number of nodes on the flowchart (which is 5 in collapsed version)?